ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -150,14 +175,110 @@ public void testDifferentMessageRetentionDays() throws StoreException, Interrupt
       MetricRegistry metricRegistry = new MetricRegistry();
       StorageManagerMetrics metrics = new StorageManagerMetrics(metricRegistry);
       blobStore = new MockBlobStore(config, metrics, time, CAPACITY_IN_BYTES, DEFAULT_USED_CAPACITY_IN_BYTES);
-      compactionManager = new CompactionManager(config, time);
+      compactionManager = new CompactionManager(MOUNT_PATH, config, Collections.singleton((BlobStore) blobStore), time);
       blobStore.validLogSegments = generateRandomStrings(2);
       verifyCompactionDetails(
           new CompactionDetails(time.milliseconds() - messageRetentionDays * Time.SecsPerDay * Time.MsPerSec,
               blobStore.validLogSegments));
     }
   }
 
+  /**
+   * Tests that compaction is triggered on all stores provided they do not misbehave. Also includes a store that is
+   * not ready for compaction. Ensures that {@link BlobStore#maybeResumeCompaction()} is called before
+   * {@link BlobStore#compact(CompactionDetails)} is called.
+   * @throws Exception
+   */
+  @Test
+  public void testCompactionExecutorHappyPath() throws Exception {
+    int numStores = 3;
+    List<BlobStore> stores = new ArrayList<>();
```

Comment:
This is needed in line 197. Also this is somewhat of a config variable and I find it more readable to put it in the beginning.